compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -539,6 +540,8 @@ trait Applications extends Compatibility {
            *           in the remaining formal parameters.
            */
           def addTyped(arg: Arg, formal: Type): List[Type] =
+            if (!elimRepeatedPhase.exists || ctx.phase <= elimRepeatedPhase) && isVarArg(arg) && !formal.isRepeatedParam then
```

Comment:
```suggestion
            if isVarArg(arg) && !formal.isRepeatedParam && !ctx.isAfterTyper then
```
We have to check this only at typer, not afterwards. So this is simpler.